libnymea-core/hardware/serialport/serialportmonitor.cpp

```diff
@@ -0,0 +1,189 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+*
+* Copyright 2013 - 2021, nymea GmbH
+* Contact: contact@nymea.io
+*
+* This file is part of nymea.
+* This project including source code and documentation is protected by
+* copyright law, and remains the property of nymea GmbH. All rights, including
+* reproduction, publication, editing and translation, are reserved. The use of
+* this project is subject to the terms of a license agreement to be concluded
+* with nymea GmbH in accordance with the terms of use of nymea GmbH, available
+* under https://nymea.io/license
+*
+* GNU Lesser General Public License Usage
+* Alternatively, this project may be redistributed and/or modified under the
+* terms of the GNU Lesser General Public License as published by the Free
+* Software Foundation; version 3. This project is distributed in the hope that
+* it will be useful, but WITHOUT ANY WARRANTY; without even the implied
+* warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this project. If not, see <https://www.gnu.org/licenses/>.
+*
+* For any further details and any questions please contact us under
+* contact@nymea.io or see our FAQ/Licensing Information on
+* https://nymea.io/license/faq
+*
+* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#include "serialportmonitor.h"
+#include "loggingcategories.h"
+
+namespace nymeaserver {
+
+NYMEA_LOGGING_CATEGORY(dcSerialPortMonitor, "SerialPortMonitor")
+
+SerialPortMonitor::SerialPortMonitor(QObject *parent) : QObject(parent)
+{
+    // Read initially all tty devices
+    foreach (const QSerialPortInfo &serialPortInfo, QSerialPortInfo::availablePorts()) {
+        m_serialPorts.insert(serialPortInfo.systemLocation(), SerialPort(serialPortInfo));
+    }
+
+#ifdef WITH_UDEV
+    // Init udev
+    m_udev = udev_new();
+    if (!m_udev) {
+        qCWarning(dcSerialPortMonitor()) << "Could not initialize udev for the adapter monitor";
+        return;
+    }
+
+    // Create udev monitor
+    m_monitor = udev_monitor_new_from_netlink(m_udev, "udev");
```

Comment:
same here. If all goes well during setup, this is never cleaned up again in the dtor or similar

Reply:
not sure why you want to delete the udev object...

Reply:
m_monitor is not deleted too, I mean.

On shutdown at latest...

Reply:
yeah, somehow the dtor has not been pushed...gonna fix this